cumulus/polkadot-parachain/polkadot-parachain-lib/src/common/types.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Cumulus.
+
+// Cumulus is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Cumulus is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
+
+use cumulus_client_consensus_common::ParachainBlockImport as TParachainBlockImport;
+use cumulus_primitives_core::relay_chain::UncheckedExtrinsic;
+use sc_consensus::DefaultImportQueue;
+use sc_executor::WasmExecutor;
+use sc_service::{PartialComponents, TFullBackend, TFullClient};
+use sc_telemetry::{Telemetry, TelemetryWorkerHandle};
+use sc_transaction_pool::FullPool;
+use sp_runtime::{generic, traits::BlakeTwo256};
+use std::sync::Arc;
+
+pub use parachains_common::{AccountId, Balance, Hash, Nonce};
+
+type Header<BlockNumber> = generic::Header<BlockNumber, BlakeTwo256>;
+pub type CustomBlock<BlockNumber> = generic::Block<Header<BlockNumber>, UncheckedExtrinsic>;
+pub type Block = CustomBlock<u32>;
+
+#[cfg(not(feature = "runtime-benchmarks"))]
+pub type ParachainHostFunctions = cumulus_client_service::ParachainHostFunctions;
```

Comment:
Sorry if this is duplicate, but can you elaborate why this is needed? 

Or you are onboard that it is only needed for the case of the hardcoded chains, not in the omni cases.

Reply:
Okay you answered in https://github.com/paritytech/polkadot-sdk/pull/5269#discussion_r1713682255